src/shogun/neuralnets/NeuralLayer.h

```diff
@@ -147,10 +148,9 @@ class CNeuralLayer : public CSGObject
 	/** Computes the activations of the neurons in this layer, results should 
 	 * be stored in m_activations. To be used only with input layers
 	 * 
-	 * @param inputs activations of the neurons in the 
-	 * previous layer, matrix of size previous_layer_num_neurons * batch_size
+	 * @param inputs dot features to store activations from
 	 */
-	virtual void compute_activations(SGMatrix<float64_t> inputs) { }
+	virtual void compute_activations(CFeatures* inputs) { }
```

Comment:
I think it would be better to keep ```inputs``` a matrix. If we gpu-ized things, and ```inputs``` was of type ```CFeatures```,  since ```CFeatures``` has its data on CPU memory, we'd have to copy it into GPU memory in every iteration of training, which would be pretty slow.